aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -237,7 +255,58 @@ def batch_processor(
     return handler(event, context)
 
 
-class BatchProcessor(BasePartialProcessor):
+@lambda_handler_decorator
+async def async_batch_processor(
+        handler: Callable, event: Dict, context: LambdaContext, record_handler: Callable, processor: AsyncBasePartialProcessor
+):
+    """
+    Middleware to handle batch event processing
+
+    Parameters
+    ----------
+    handler: Callable
+        Lambda's handler
+    event: Dict
+        Lambda's Event
+    context: LambdaContext
+        Lambda's Context
+    record_handler: Callable
+        Callable to process each record from the batch
+    processor: BasePartialProcessor
+        Batch Processor to handle partial failure cases
+
+    Examples
+    --------
+    **Processes Lambda's event with a BasePartialProcessor**
+
+        >>> from aws_lambda_powertools.asynchrony import async_lambda_handler
+        >>> from aws_lambda_powertools.utilities.batch import async_batch_processor, AsyncBatchProcessor
+        >>>
+        >>> async def async_record_handler(record):
+        >>>     payload: str = record.body
+        >>>     return payload
+        >>>
+        >>> processor = AsyncBatchProcessor(event_type=EventType.SQS)
+        >>>
+        >>> @async_lambda_handler
+        >>> @async_batch_processor(record_handler=async_record_handler, processor=processor)
+        >>> async def lambda_handler(event, context: LambdaContext):
+        >>>     return processor.response()
+
+    Limitations
+    -----------
+    * Async batch processors
+
+    """
+    records = event["Records"]
+
+    with processor(records, record_handler, lambda_context=context):
+        await processor.async_process()
+
+    return await handler(event, context)
```

Comment:
I've been thinking about this for the last hour, please feel free to correct me if I misread it.

I think we need to focus on **`processor`** being async while keeping the Lambda Handler synchronous. 

This means transforming **`async_batch_processor`** to **`def async_batch_processor`**, and use **`asyncio.run`** to prime and wait for the **`processor.async_process()`** coroutine to complete. Another benefit is making `async_lambda_handler` redundant.


**Why is that?**

1. **Changing the Lambda Handler to async requires an additional decorator.** This introduces an unnecessary overhead with multiple function calls, considering we have a decorator factory here (3fn) twice + decorated functions. This slows execution down, including [`trace_execution`](https://awslabs.github.io/aws-lambda-powertools-python/2.3.1/utilities/middleware_factory/#tracing-middleware-execution) feature will not work as expected and **might** block the event loop.
2. **Unforeseen issues with decorator stacking**. Customers often use other synchronous decorators like `@metrics.log_metrics`, `@logger`, `@tracer`,  `@validate`, and a mix of their own. A common issue is ordering and I'm 90% confident someone will put this in the wrong handler, for the Lambda handler.